Wrappers/Python/cil/framework/acquisition_geometry.py

```diff
@@ -0,0 +1,1939 @@
+                                          pixel_offset,
+                                          direction='horizontal'):
+        """
+        Adjusts the position of the detector in a specified direction within the imaging plane.
+
+        Parameters:
+        -----------
+        pixel_offset : float
+            The number of pixels to adjust the detector's position by.
+        direction : {'horizontal', 'vertical'}, optional
+            The direction in which to adjust the detector's position. Defaults to 'horizontal'.
+
+        Notes:
+        ------
+        - If `direction` is 'horizontal':
+            - If the panel's origin is 'left', positive offsets translate the detector to the right.
+            - If the panel's origin is 'right', positive offsets translate the detector to the left.
+
+        - If `direction` is 'vertical':
+            - If the panel's origin is 'bottom', positive offsets translate the detector upward.
+            - If the panel's origin is 'top', positive offsets translate the detector downward.
+
+        Returns:
+        --------
+        None
+        """
+
+        if direction == 'horizontal':
+            pixel_size = self.panel.pixel_size[0]
+            pixel_direction = self.system.detector.direction_x
+
+        elif direction == 'vertical':
+            pixel_size = self.panel.pixel_size[1]
+            pixel_direction = self.system.detector.direction_y
+
+        if 'bottom' in self.panel.origin or 'left' in self.panel.origin:
+            self.system.detector.position -= pixel_offset * pixel_direction * pixel_size
+        else:
+            self.system.detector.position += pixel_offset * pixel_direction * pixel_size
+
+
+    def __str__(self):
+        repres = ""
+        if self.configured:
+            repres += str(self.system)
+            repres += str(self.panel)
+            repres += str(self.channels)
+            repres += str(self.angles)
+
+            repres += "Distances in units: {}".format(self.units)
+
+        return repres
+
+    def __eq__(self, other):
+
+        if not isinstance(other, self.__class__):
+            return False
+
+        if self.system == other.system\
+        and self.panel == other.panel\
+        and self.channels == other.channels\
+        and self.angles == other.angles:
+            return True
+
+        return False
+
+
+class AcquisitionGeometry(BaseAcquisitionGeometry):
+    """This class holds the AcquisitionGeometry of the system.
+
+    Please initialise the AcquisitionGeometry using the using the static methods:
+
+    `AcquisitionGeometry.create_Parallel2D()`
+
+    `AcquisitionGeometry.create_Cone2D()`
+
+    `AcquisitionGeometry.create_Parallel3D()`
+
+    `AcquisitionGeometry.create_Cone3D()`
+    """
+
+
+    #for backwards compatibility
+    @property
+    def ANGLE(self):
+        warnings.warn("use acquisition_labels['ANGLE'] instead", DeprecationWarning, stacklevel=2)
+        return acquisition_labels['ANGLE']
```

Comment:
Should there also be a similar property for RANDOM and others listed in acquisition_labels which used to be properties of AcquisitionGeometry?

Reply:
there were no instances of `AcquisitionGeometry.RANDOM` afaik.